### PR TITLE
Galoshes no longer runtime when dispensed from a vendor.

### PR DESCRIPTION
**Pull request by Adrer**
## What Does This PR Do
Galoshes no longer cause a runtime trying to get an item slot from a user that doesn't exist.

Fixes: #24163

## Why It's Good For The Game
Runtime bad.

## Testing
Dispense galoshes from a janitor vendor with my hands full
No runtime baby.

## Changelog
NPFC

### CODE REVIEW

File: code/modules/clothing/clothing.dm [hunk 1]
```diff
@@ -492,7 +492,7 @@
 	if(!no_slip)
 		return
 	var/mob/living/carbon/human/H = user
```
Comment (Henri215) [inline code/modules/clothing/clothing.dm:494]: Is it not better to do that check before it receives the user?

Comment (DGamerL) [reply code/modules/clothing/clothing.dm:494]: Yeah just do an
```
if(!user)
      return
```
Above the typecast

File: code/modules/clothing/clothing.dm [hunk 2]
```diff
@@ -492,6 +492,8 @@
...
+	if(!user)
+    	return
```
Comment (1080pCat) [inline code/modules/clothing/clothing.dm:496]:
```suggestion
        return
```

Comment (1080pCat) [reply code/modules/clothing/clothing.dm:496]: gonna hope that doesn't do the exact same thing, indenting issues

Comment (1080pCat) [inline code/modules/clothing/clothing.dm:496]: Using spaces as indents, vcs and git is CRINGE and wont let suggested changes fix it

**Review by 1080pCat — CHANGES_REQUESTED** [review]

**Review by lewcc — APPROVED** [review]

**Pull request merged**